README.md

```diff
@@ -38,15 +54,50 @@ use ItkDev\OpenIdConnect\Security\OpenIdConfigurationProvider;
 
 $provider = new OpenIdConfigurationProvider([
     'redirectUri' => 'https://some.url', // Absolute url to where the user is redirected after a successful login            
-    'urlConfiguration' => 'https:/.../openid-configuration', // url to OpenId Discovery document
-    'cachePath' => '/some/directory/openId-cache.php', // Path for caching above discovery document
+    'openIDConnectMetadataUrl' => 'https:/.../openid-configuration', // url to OpenId Discovery document
+    'cacheItemPool' => 'Psr6/CacheItemPoolInterface', // Pool for caching above discovery document
     'clientId' => 'client_id', // Client id assigned by authorizer
     'clientSecret' => 'client_secret', // Client password assigned by authorizer
  ]);
+ ```
+
+#### Non-authorized requests
+
+Non-authorized requests should be redirected to the authorization url.
+
+To generate the authorization url you must supply "state" and "nonce":
 
-$authUrl = $provider->getAuthorizationUrl();
+State:  
+_"A value included in the request that's also returned in the token response.
+It can be a string of any content that you want. A randomly generated unique
+value is typically used for preventing cross-site request forgery attacks.
+The state is also used to encode information about the user's state in the
+application before the authentication request occurred, such as the page they
+were on."_
 
-// direct to $authUrl
+Nonce:  
+_"A value included in the request (generated by the application) that is
+included in the resulting ID token as a claim. The application can then verify
+this value to mitigate token replay attacks. The value is typically a randomized
+unique string that can be used to identify the origin of the request."_
+
+See: [Send authentication requests](https://docs.microsoft.com/en-us/azure/active-directory-b2c/openid-connect#send-authentication-requests)
+
+You must persist these locally so that they can be used to validate the token
+when the user is redirected back to your application.
+
+```php
+// Get "state" and "nonce"
+$state = $provider->getState();
```

Comment:
Isn't  "state" only given a value once we have called 'getAuthorizationUrl'?

Reply:
Good catch. Changed to use `getRandomState()`

Reply:
`getRandomState()` is protected in AbstractProvider, possibly add `generateState()` function similar to  `generateNonce()`?

Reply:
Thank you. Done.